src/evt/player_api.c

```diff
@@ -449,32 +841,95 @@ ApiStatus PlaySoundAtPlayer(ScriptInstance* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "evt/player_api", func_802D2D30);
+void func_802D2D30(u8 r, u8 g, u8 b, u8 a, u16 left, u16 top, u16 right, u16 bottom) {
+    gDPPipeSync(gMasterGfxPos++);
 
-INCLUDE_ASM(s32, "evt/player_api", func_802D2ED4);
+    if (a == 0xFF) {
+        gDPSetCombineLERP(gMasterGfxPos++, 0, 0, 0, PRIMITIVE, 0, 0, 0, 1, 0, 0, 0, PRIMITIVE, 0, 0, 0, 1);
+    } else {
+        gDPSetRenderMode(gMasterGfxPos++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+        gDPSetCombineMode(gMasterGfxPos++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+    }
 
-INCLUDE_ASM(s32, "evt/player_api", func_802D2F34);
+    gDPSetPrimColor(gMasterGfxPos++, 0, 0, r, g, b, a);
+    gDPFillRectangle(gMasterGfxPos++, left, top, right, bottom);
+    gDPPipeSync(gMasterGfxPos++);
+    gDPSetRenderMode(gMasterGfxPos++, G_RM_TEX_EDGE, G_RM_TEX_EDGE2);
+    gDPSetCombineMode(gMasterGfxPos++, G_CC_DECALRGBA, G_CC_DECALRGBA);
+}
+
+void func_802D2ED4(s32 arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, s32 arg5, s32 arg6, s32 arg7) {
+    u16 temp1 = arg4 + arg6;
+    u16 temp2 = arg5 + arg7;
 
-INCLUDE_ASM(s32, "evt/player_api", func_802D2FCC);
+    func_802D2D30(arg0, arg1, arg2, arg3, arg4, arg5, temp1, temp2);
+}
 
-INCLUDE_ASM(s32, "evt/player_api", func_802D3028);
+void func_802D2F34(VirtualEntity* virtualEntity, f32 arg1, f32 arg2) {
+    f32 theta = (arg2 * TAU) / 360.0f;
+    f32 sinTheta = sin_rad(theta);
+    f32 cosTheta = cos_rad(theta);
+
+    virtualEntity->pos.x += arg1 * sinTheta;
+    virtualEntity->pos.z += -arg1 * cosTheta;
+}
+
+void func_802D2FCC(void) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(*D_802DB7C0); i++) {
+        VirtualEntity* virtualEntity = (*D_802DB7C0)[i];
+
+        if (virtualEntity != NULL && virtualEntity->entityModelIndex >= 0) {
+            exec_entity_model_commandlist(virtualEntity->entityModelIndex);
+        }
+    }
+}
+
+INCLUDE_ASM(void, "evt/player_api", func_802D3028, void);
 
 INCLUDE_ASM(s32, "evt/player_api", func_802D31E0);
 
-INCLUDE_ASM(s32, "evt/player_api", func_802D3398);
+ApiStatus func_802D3398(ScriptInstance* script, s32 isInitialCall) {
+    if (!gGameStatusPtr->isBattle) {
+        D_802DB7C0 = &D_802DB6C0;
```

Comment:
rename these symbols maybe?
D_802DB7C0 == gVirtualEntityListPtr
D_802DB6C0 == gBattleVirtualEntityList
D_802DB5C0 == gWorldVirtualEntityList
or so.

Reply:
gonna resolve this because we plan to do an automatic symbol name merge soon, and I'd rather do that before changing more names